fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexPrefetchTest.java

```diff
@@ -0,0 +1,483 @@
+        });
+        // Second iteration - last 5 records
+        continuation = executeAndVerifyData(plan, continuation, executeProperties, 5, (rec, i) -> {
+            int primaryKey = 4 - i;
+            String strValue = ((primaryKey % 2) == 0) ? "even" : "odd";
+            int numValue = 1000 - primaryKey;
+            assertRecord(rec, primaryKey, strValue, numValue, "MySimpleRecord$num_value_unique", (long)numValue, primaryKey);
+        });
+        // Third iteration - no more values to read
+        continuation = executeAndVerifyData(plan, continuation, executeProperties, 0, (rec, i) -> {
+        });
+        assertNull(continuation);
+    }
+
+    /*
+     * Test continuation where the continued plan uses a different prefetch mode than the original plan.
+     */
+    @Test
+    void indexPrefetchWithMixedContinuationTest() throws Exception {
+        RecordQueryPlan planWithPrefetch = plan(NUM_VALUES_LARGER_THAN_990, RecordQueryPlannerConfiguration.IndexPrefetchUse.USE_INDEX_PREFETCH);
+        RecordQueryPlan planWithScan = plan(NUM_VALUES_LARGER_THAN_990, RecordQueryPlannerConfiguration.IndexPrefetchUse.NONE);
+        ExecuteProperties executeProperties = ExecuteProperties.newBuilder()
+                .setReturnedRowLimit(4)
+                .build();
+
+        // First iteration - first 4 records
+        byte[] continuation = executeAndVerifyData(planWithPrefetch, null, executeProperties, 4, (rec, i) -> {
+            int primaryKey = 9 - i;
+            String strValue = ((primaryKey % 2) == 0) ? "even" : "odd";
+            int numValue = 1000 - primaryKey;
+            assertRecord(rec, primaryKey, strValue, numValue, "MySimpleRecord$num_value_unique", (long)numValue, primaryKey);
+        });
+        // Second iteration - second 4 records
+        continuation = executeAndVerifyData(planWithScan, continuation, executeProperties, 4, (rec, i) -> {
+            int primaryKey = 5 - i;
+            String strValue = ((primaryKey % 2) == 0) ? "even" : "odd";
+            int numValue = 1000 - primaryKey;
+            assertRecord(rec, primaryKey, strValue, numValue, "MySimpleRecord$num_value_unique", (long)numValue, primaryKey);
+        });
+        // Third iteration - last 2 records
+        continuation = executeAndVerifyData(planWithPrefetch, continuation, executeProperties, 2, (rec, i) -> {
+            int primaryKey = 1 - i;
+            String strValue = ((primaryKey % 2) == 0) ? "even" : "odd";
+            int numValue = 1000 - primaryKey;
+            assertRecord(rec, primaryKey, strValue, numValue, "MySimpleRecord$num_value_unique", (long)numValue, primaryKey);
+        });
+        assertNull(continuation);
+    }
+
+    @ParameterizedTest(name = "indexPrefetchByteLimitContinuation(" + ARGUMENTS_WITH_NAMES_PLACEHOLDER + ")")
+    @EnumSource()
+    @Disabled // This test is inconsistently failing (for the NONE case)
```

Comment:
Is it failing because in the `NONE` case because of differences in the way the byte limit is handled?

Reply:
I believe it is failing because of some byte counting that happens sometimes. My suspicion is that writing some metadata may count and throw the test math off. The tests succeed consistently when run in isolation but sometimes break when run  as part of a complete build.